packages/related-keyphrase-suggestions/src/components/KeyphrasesTable/index.js

```diff
@@ -0,0 +1,193 @@
+			</Table.Cell>
+			<Table.Cell>
+				<div className="yst-flex yst-gap-2">
+					<SkeletonLoader className="yst-w-4 yst-h-5" />
+					<SkeletonLoader className="yst-w-3 yst-h-5" />
+				</div>
+			</Table.Cell>
+			{ withButton &&
+			<Table.Cell>
+				<SkeletonLoader className="yst-w-16 yst-h-7" />
+			</Table.Cell>
+			}
+		</Table.Row>
+	);
+};
+
+LoadingKeyphrasesTableRow.propTypes = {
+	withButton: PropTypes.bool,
+};
+
+const intentMapping = [ "i", "n", "t", "c" ];
+
+/**
+ * Prepare the row data.
+ * @param {array} columnNames The column names.
+ * @param {array} row The row data.
+ * @returns {object} The prepared row.
+ */
+const prepareRow = ( columnNames, row ) => {
+	const rowData = {};
+	columnNames.forEach( ( columnName, index ) => {
+		switch ( columnName ) {
+			case "Trends":
+				rowData.trends = row[ index ].split( "," ).map( ( value ) => parseFloat( value ) );
+				break;
+			case "Intent":
+				rowData.intent = row[ index ].split( "," ).map( ( value ) => intentMapping[ Number( value ) ] );
+				break;
+			case "Keyword Difficulty Index":
+				rowData.keywordDifficultyIndex = Number( row[ index ] );
+				break;
+			case "Search Volume":
+				rowData.searchVolume = row[ index ];
+				break;
+			default:
+				rowData[ columnName.toLowerCase() ] = row[ index ];
+		}
+	} );
+	return rowData;
+};
+
+const headers = [
+	__( "Related keyphrase", "wordpress-seo" ),
+	__( "Intent", "wordpress-seo" ),
+	__( "Volume", "wordpress-seo" ),
+	__( "Trend", "wordpress-seo" ),
+	__( "Difficulty %", "wordpress-seo" ),
+];
+
+/**
+ *
+ * @param {object} data The rows to display in the table.
+ * @param {Function} renderButton The render button function.
+ * @param {Array} relatedKeyphrases The related keyphrases.
+ *
+ * @returns  {JSX.Element} The keyphrases table.
+ */
+export const KeyphrasesTable = ( { data, renderButton, relatedKeyphrases } ) => {
+	const rows = data?.results?.rows?.map( row => prepareRow(  data.results.columnNames, row ) );
+
+	return <Table>
+		<Table.Head>
+			<Table.Row>
+				{ headers && headers.map( ( headerLabel, index ) =>
+					<Table.Header key={ `header-${ index }` }>
+						{ headerLabel }
+					</Table.Header> )
+				}
+
+				{ renderButton && <Table.Header className="yst-text-right yst-w-20">
+					{ __( "Add keyphrase", "wordpress-seo" ) }
+				</Table.Header> }
+
+			</Table.Row>
+		</Table.Head>
+
+		<Table.Body>
+
+			{ rows ? rows.map( ( rowData, index ) =>
+				<KeyphrasesTableRow
+					key={ `related-keyphrase-${ index }` }
+					renderButton={ renderButton }
+					relatedKeyphrases={ relatedKeyphrases }
+					{ ...rowData }
+				/>,
+			)
+			// Show 10 loading rows when there are no rows.
+				: Array.from( { length: 10 }, ( _, index ) =>
+					<LoadingKeyphrasesTableRow key={ `loading-row-${ index }` } withButton={ ! isUndefined( renderButton ) }  /> )
+			}
+		</Table.Body>
+	</Table>;
+};
+
+KeyphrasesTable.propTypes = {
+	data: PropTypes.object,
```

Comment:
We only look at the `results` in there (`rows` and `columnNames`. Why add that layer to the API of this component?
Seems like you can split those even.